frontend/views/containers/proposals/ProposalTemplate.vue

```diff
@@ -104,18 +135,20 @@ export default {
     },
     submit () {
       this.next()
-      this.$emit('submit')
+      this.$emit('submit', {
+        reason: this.$refs.reason.value
+      })
     }
   },
   computed: {
     isNextStep () {
-      return this.currentStep < this.maxSteps - 1
+      return this.currentStep <= this.maxSteps - 1
     },
-    isLastStep () {
-      return this.currentStep === this.maxSteps - 1
+    isReasonStep () {
+      return this.currentStep === this.maxSteps
     },
     isConfirmation () {
-      return this.currentStep === this.maxSteps
+      return this.currentStep === this.maxSteps + 1
```

Comment:
I'm guessing there's a cleaner and less bug-prone way of doing this?

What happens if the user keeps pressing next/submit? I haven't tested this locally yet... so maybe that will answer my question, but looking at the code I feel like this isn't usually how these bounds checks are done...

Reply:
At the moment I'm not seeing another way of approaching this. Can we keep it for now and iterate in a next PR related to proposals? (there will be a bunch of them!)

Reply:
Sure